### PR TITLE
Bug 1533835 - New Library shortcuts for Firefox Home (TopSites)

### PR DESCRIPTION
**Pull request by farhanpatel**
This code is disabled on the iPad. We'll add a new Library button to the URLBar there.

### CODE REVIEW

File: Client/Frontend/Strings.swift [hunk 1]
```diff
@@ -506,7 +506,7 @@ extension Strings {
 
 // App menu.
 extension Strings {
-    public static let AppMenuLibraryTitleString = NSLocalizedString("Menu.Library.Title", tableName: "Menu", value: "Library", comment: "Label for the button, displayed in the menu, used to open the Library")
+    public static let AppMenuLibraryTitleString = NSLocalizedString("Menu.Library.Title", tableName: "Menu", value: "Your Library", comment: "Label for the button, displayed in the menu, used to open the Library")
```
Comment (farhanpatel) [inline Client/Frontend/Strings.swift]: I hacked this change into the last string export. So this should be okay 😉

**Comment by garvankeeley** [comment]
Orientation change layout doesn't look right, and the title colors aren't adapting to dark mode:
image

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 2]
```diff
@@ -23,6 +23,9 @@ struct ASPanelUX {
     static let SectionInsetsForIpad: CGFloat = 101
     static let SectionInsetsForIphone: CGFloat = 14
     static let MinimumInsets: CGFloat = 14
+    static let LibraryShortcutsHeight: CGFloat = 100
+    static let LibraryShortcutsMaxWidth: CGFloat = 350
+    static let PocketRed = UIColor(rgb: 0xEF4156)
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: Probably is ok to not be themeable, but perhaps should be a Photon color?

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 3]
```diff
@@ -277,20 +288,32 @@ extension ActivityStreamPanel: UICollectionViewDelegateFlowLayout {
...
+                    view.iconView.tintColor = ASPanelUX.PocketRed
                     return view
                 case .topSites:
-                    return UICollectionReusableView()
+                    view.title = title
+                    view.moreButton.isHidden = true
+                    return view
+                case .libraryShortcuts:
+                    view.title = title
+                    view.moreButton.isHidden = true
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: append this `case` to the above `case`, IIUC, they are the same

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 4]
```diff
@@ -314,6 +337,8 @@ extension ActivityStreamPanel: UICollectionViewDelegateFlowLayout {
             return CGSize(width: cellSize.width, height: estimatedLayout.size.height)
         case .pocket:
             return cellSize
+        case .libraryShortcuts:
+            return CGSize(width: min(ASPanelUX.LibraryShortcutsMaxWidth, cellSize.width), height: (cellSize.width/4)+10)
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: nit: spacing on operators. Also comment on what div 4 plus 10 is (or perhaps there is a constant that the 4 is related to?)

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 5]
```diff
@@ -810,22 +887,33 @@ class ASHeaderView: UICollectionReusableView {
         moreButton.isHidden = true
         titleLabel.text = nil
         moreButton.removeTarget(nil, action: nil, for: .allEvents)
+        iconView.isHidden = true
+        iconView.tintColor =  UIColor.theme.homePanel.activityStreamHeaderText
+        titleLabel.textColor = UIColor.theme.homePanel.activityStreamHeaderText
+        moreButton.setTitleColor(UIConstants.SystemBlueColor, for: .normal)
+
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: nit: newline

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 6]
```diff
@@ -840,6 +928,97 @@ class ASHeaderView: UICollectionReusableView {
...
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func layoutSubviews() {
+        button.layer.cornerRadius = self.frame.width/2
+        super.layoutSubviews()
+    }
+}
+
+
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: nit: i think 2 newlines triggers lint warning

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 7]
```diff
@@ -840,6 +928,97 @@ class ASHeaderView: UICollectionReusableView {
...
+        button.contentHorizontalAlignment = .fill
+        button.imageEdgeInsets = UIEdgeInsets(equalInset: 15)
+    }
+
+    required init(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func layoutSubviews() {
+        button.layer.cornerRadius = self.frame.width/2
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: nit: spacing on operators

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 8]
```diff
@@ -840,6 +928,97 @@ class ASHeaderView: UICollectionReusableView {
...
+
+class ASLibraryCell: UICollectionViewCell {
+
+    var mainView = UIStackView()
+
+    struct LibraryPanel {
+        let title: String
+        let image: UIImage?
+        let color: UIColor
+        //let selector: Selector
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: nit: remove?

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 9]
```diff
@@ -840,6 +928,97 @@ class ASHeaderView: UICollectionReusableView {
...
+    }
+
+    var libraryButtons: [LibraryShortcutView] = []
+
+    let bookmarks = LibraryPanel(title: "Bookmarks", image: UIImage.templateImageNamed("menu-Bookmark"), color: UIColor(rgb: 0x0A84FF))
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: Should this be a themable color?

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 10]
```diff
@@ -840,6 +928,97 @@ class ASHeaderView: UICollectionReusableView {
...
+        mainView.spacing = 25
+        addSubview(mainView)
+        mainView.snp.makeConstraints { make in
+            make.edges.equalTo(self).inset(UIEdgeInsets(top: 0, left: 5, bottom: 0, right: 5))
+        }
+
+        [bookmarks,history,readingList, downloads].forEach { item in
+            let view = LibraryShortcutView()
+            view.button.setImage(item.image, for: .normal)
+            view.title.text = item.title
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: view.title set color to a UIColor.theme.somecolor ?

File: Client/Frontend/Home/ActivityStreamPanel.swift (hunk 9 above)
Comment (farhanpatel) [reply Client/Frontend/Home/ActivityStreamPanel.swift]: These colors arent 100% I think leaving them here for now is fine. I havent gotten dark theme colors for these buttons yet.

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 11]
```diff
@@ -338,7 +332,9 @@ extension ActivityStreamPanel: UICollectionViewDelegateFlowLayout {
         case .pocket:
             return cellSize
         case .libraryShortcuts:
-            return CGSize(width: min(ASPanelUX.LibraryShortcutsMaxWidth, cellSize.width), height: (cellSize.width/4)+10)
+            let numberofshortcuts: CGFloat = 4
+            let titleSpacing: CGFloat = 10
+            return CGSize(width: min(ASPanelUX.LibraryShortcutsMaxWidth, cellSize.width), height: (cellSize.width/numberofshortcuts) + titleSpacing)
```
Comment (garvankeeley) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: teeny nit: on `/` I think swift lint will complain about lack of space around math operators

**Pull request merged**